src/rgw/rgw_user.cc

```diff
@@ -2781,6 +2781,7 @@ class RGWUserMetadataHandler : public RGWMetadataHandler {
     int ret = store->list_raw_objects_init(store->get_zone_params().user_uid_pool, marker,
                                            &info->ctx);
     if (ret < 0) {
+      delete info;
```

Comment:
thanks, good catch! i'm fine with this fix as is it, but we might consider taking it step further with `std::unique_ptr`:
```c++
auto info = ceph::make_unique<list_keys_info>();

...

*phandle = (void *)info.release();
```
the only advantage here being exception safety in the event that `store->list_raw_objects_init()` throws

Reply:
@cbodley You mean we can handle it like this:
```
     auto info = ceph::make_unique<list_keys_info>();
     ...
     int ret = store->list_raw_objects_init(store->get_zone_params().domain_root, marker,
                                            &info->ctx);
     if (ret < 0) {
       *phandle = (void *)info.release();
       return ret;
     }
     *phandle = (void *)info;
```
So that we can handle `store->list_raw_objects_init` throws safety.
Did I catch the point?

Reply:
@cbodley That's a good point. Make it `std::unique_ptr` and let it delete itself when it goes out of scope / throws / returns.

Reply:
>   *phandle = (void *)info.release();

@scienceluo  I think this will again leak.

Reply:
@joscollin So what's the exact way to solve this?

Reply:
```
auto info = ceph::make_unique<list_keys_info>();

int ret = store->list_raw_objects_init(store->get_zone_params().domain_root, marker, &info->ctx);
if (ret < 0) {
  //Let info delete the object when the function returns.
  return ret;
}

*phandle = (void *)info.release(); //Release the ownership and assign it to phandle for output.
```

I think this is correct, as per @cbodley. @cbodley Could you please verify ? Thanks.

Reply:
@cbodley ping.

Reply:
@joscollin I found some `delete` case like this in other functions, maybe we can open another PR to fix this like @cbodley and you said.

Reply:
@scienceluo Yes, you can do that. But let's confirm this fix first.

Reply:
thanks @scienceluo @joscollin. looks good, but the comments are a bit redundant

Reply:
@cbodley @joscollin Thanks, and the comment has been removed.

Reply:
> but the comments are a bit redundant

@cbodley Yes, I have put the comments only to explain the change to @scienceluo.

Sorry for the late response, I was on leave.